#+title: bcbio validation: build 38, low frequency somatic variants, structural variations
#+author: Brad Chapman \newline Bioinformatics Core, Harvard Chan School \newline https://bcb.io \newline http://j.mp/bcbiolinks
#+date: 5 November 2015

#+OPTIONS: toc:nil H:2

#+startup: beamer
#+LaTeX_CLASS: beamer
#+latex_header: \usepackage{url}
#+latex_header: \usepackage{hyperref}
#+latex_header: \hypersetup{colorlinks=true}
#+BEAMER_THEME: default
#+BEAMER_COLOR_THEME: seahorse
#+BEAMER_INNER_THEME: rectangles

* Build 38

** Outline

\Large
- \textbf{Human build 38}
- Low frequency somatic calling
- Structural variation

** Currently: GRCh37/hg19

#+ATTR_LATEX: :width .9\textwidth
[[./images8/GRCh37.jpg]]

\small
http://www.ncbi.nlm.nih.gov/books/NBK153600/?report=reader

** GRCh38 -- graph based, many more alternative loci

#+ATTR_LATEX: :width .9\textwidth
[[./images8/GRCh38.jpg]]

\footnotesize
http://www.slideshare.net/GenomeRef/transitioning-to-grch38

** GRCh38 -- advantage for variant calling

#+ATTR_LATEX: :width .9\textwidth
[[./images8/GRCh38-advantages.jpg]]

\footnotesize
http://www.slideshare.net/GenomeRef/transitioning-to-grch38

** Avoiding collapsed repeats

#+ATTR_LATEX: :width 0.8\textwidth
[[./images7/hg38_collapsed_repeats.jpg]]

\tiny
http://www.slideshare.net/kmsteinberg/the-importance-of-high-quality-reference-genome-assemblies-to-personal-and-medical-genomics

** Comparison

\Large
- Build 37 and 38
- Validation sets: Genome in a Bottle, Illumina Platinum Genomes
- Lift-over methods: CrossMap/LiftOver, NCBI Remap
- 38 builds: with/without alternative alleles
- Variant callers: FreeBayes, GATK HaplotypeCaller

\normalsize
http://bcb.io/2015/09/17/hg38-validation/

** Reference materials

#+BEGIN_CENTER
#+ATTR_LATEX: :width .5\textwidth
[[./images/giab.png]]

#+ATTR_LATEX: :width .7\textwidth
[[./images7/ga4gh.png]]

#+ATTR_LATEX: :width .9\textwidth
[[./images7/dream_challenge.png]]
#+END_CENTER

[[http://www.genomeinabottle.org/]]
http://ga4gh.org/#/benchmarking-team
https://www.synapse.org/#!Synapse:syn312572

** 

#+BEGIN_CENTER
#+ATTR_LATEX: :width .9\textwidth
[[./images7/hg38_val.png]]
#+END_CENTER

** 

#+BEGIN_CENTER
#+ATTR_LATEX: :width .8\textwidth
[[./images7/hg38_val_giab.png]]
#+END_CENTER

** Small variant results

\Large
- SNPs: build 38 more sensitive
- SNPs: build 38 reduces false positives
- Indels: build 38 detected more
- Indels: work on sensitivity and precision

** Remapping results

\Large
Need conversion approaches for resources not yet available on build 38

\Large
- CrossMap: \newline
  \normalsize http://crossmap.sourceforge.net/
\Large
- NCBI remap: \newline
  \normalsize http://www.ncbi.nlm.nih.gov/genome/tools/remap
\Large
- Both performed well
- NCBI remap has additional sensitivity, but needs tuning

** Major histocompatibility complex (MHC) -- HLAs

[[./images8/MHC.png]]

\small
\vspace{1cm}
http://www.ebi.ac.uk/ipd/imgt/hla/ \\
\scriptsize
http://sciscogenetics.com/technology/human-leukocyte-antigen-complex/

** Alignment: bwa alternative allele support

#+ATTR_LATEX: :width .9\textwidth
[[./images8/bwa-alts.png]]

\small
https://github.com/lh3/bwa/blob/master/README-alt.md

** HLA typing

\Large
- bwakit implementation
- 1000 genomes: build 38 + IMGT/HLA-3.18.0
- bwa extracts HLA reads
- fermi de novo assembly
- Remap assemblies back to HLA choices
- Call HLA types

\footnotesize
https://github.com/lh3/bwa/blob/master/README-alt.md#hla-typing

** HLA nomenclature

#+ATTR_LATEX: :width .9\textwidth
[[./images7/hla_naming.png]]

https://www.ebi.ac.uk/ipd/imgt/hla/
http://hla.alleles.org/alleles/p_groups.html

** Validations

\Large
- Omixon example data
- bwakit calls on exome and deep targeted data
- P-group resolution
- Good results for exome
- Assembly problems with deep targeted

\footnotesize
http://www.omixon.com/hla-typing-example-data/
https://gist.github.com/chapmanb/8e2a18c7bbbee3167395

** Genome graphs and variation

#+ATTR_LATEX: :width .9\textwidth
[[./images8/discovar-graphs.jpg]]

\footnotesize
http://www.nature.com/ng/journal/v46/n12/fig_tab/ng.3121_SF6.html
\normalsize

** vg -- tools for working with variant graphs

#+BEGIN_CENTER
#+ATTR_LATEX: :width .5\textwidth
[[./images8/vg-graphs.png]]
#+END_CENTER

\tiny
https://docs.google.com/presentation/d/1bbl2zY4qWQ0yYBHhoVuXb79HdgajRotIUa_VEn3kTpI
\normalsize

* Somatic calling

** Outline

\Large
- Human build 38
- \textbf{Low frequency somatic calling}
- Structural variation

** Cancer somatic calling

[[./images8/cancer-somatic.png]]

\vspace{0.5cm}
\scriptsize
http://www.nature.com/nmeth/journal/v10/n8/fig_tab/nmeth.2562_F1.html
\normalsize

** Cancer heterogeneity

[[./images8/Treatment_bottleneck.pdf]]

\small
http://en.wikipedia.org/wiki/Tumour_heterogeneity
\normalsize

** VarDict

\Large
- AstraZeneca
- SNP + Insertion/Deletions
- Works on very deep targeted data

\vspace{1cm}
\small
https://github.com/AstraZeneca-NGS/VarDictJava
\normalsize

** DREAM synthetic dataset 4

#+BEGIN_CENTER
#+ATTR_LATEX: :width 1.0\textwidth
[[./images7/dream_syn4.png]]
#+END_CENTER

\footnotesize
https://www.synapse.org/#!Synapse:syn312572/wiki/62018

** VarDict sensivitity/precision before

#+BEGIN_CENTER
#+ATTR_LATEX: :width 1.0\textwidth
[[./images7/vardict_val_orig.png]]
#+END_CENTER

** VarDict sensivitity/precision after

#+BEGIN_CENTER
#+ATTR_LATEX: :width 1.0\textwidth
[[./images7/vardict_val.png]]
#+END_CENTER

** How? Filter summary

\Large
#+BEGIN_SRC sh
((AF * DP < 6) &&
 ((MQ < 55.0 && NM > 1.0) ||
  (MQ < 60.0 && NM > 2.0) ||
  (DP < 10) ||
  (QUAL < 45)))
#+END_SRC

** Filter: mapping quality and number of mismatches

#+BEGIN_CENTER
#+ATTR_LATEX: :width 1.0\textwidth
[[./images7/vardict_filter_nmmq.png]]
#+END_CENTER

** Filter: low depth

#+BEGIN_CENTER
#+ATTR_LATEX: :width 1.0\textwidth
[[./images7/vardict_filter_dp.png]]
#+END_CENTER

** Filter: low quality

#+BEGIN_CENTER
#+ATTR_LATEX: :width 1.0\textwidth
[[./images7/vardict_filter_qual.png]]
#+END_CENTER

** How can we improve?

\Large
- Incorporate machine learning methods
- Generalize with additional datasets
- AML31: http://aml31.genome.wustl.edu/

* Structural variation

** Outline

\Large
- Human build 38
- Low frequency somatic calling
- \textbf{Structural variation}

** Structural variants critical in cancer

#+ATTR_LATEX: :width 1.0\textwidth
[[./images9/nature14169-sf2.jpg]]

\scriptsize
http://www.nature.com/nature/journal/v518/n7540/full/nature14169.html

** Improvements in speed, sensitivity and precision

- \Large Lumpy: \footnotesize https://github.com/arq5x/lumpy-sv
- \Large Manta: \footnotesize https://github.com/Illumina/manta
- \Large CNVkit: \footnotesize https://github.com/etal/cnvkit
- \Large WHAM: \footnotesize https://github.com/zeeev/wham
- \Large MetaSV: \footnotesize https://github.com/bioinform/metasv

** Last year: Somatic deletions

#+BEGIN_CENTER
#+ATTR_LATEX: :width 0.9\textwidth
[[./images7/sv_dream_old.png]]
#+END_CENTER

** Results: Germline deletions

#+BEGIN_CENTER
#+ATTR_LATEX: :width 0.8\textwidth
[[./images7/sv_giab_del.png]]
#+END_CENTER

** Results: Somatic deletions

#+BEGIN_CENTER
#+ATTR_LATEX: :width 0.9\textwidth
[[./images7/sv_dream_del.png]]
#+END_CENTER

** Results: Somatic insertions

#+BEGIN_CENTER
#+ATTR_LATEX: :width 0.9\textwidth
[[./images7/sv_dream_ins.png]]
#+END_CENTER

** Prioritize in previously known regions

#+ATTR_LATEX: :width 1.0\textwidth
[[./images9/prioritize-overview.png]]

** Public cancer variant databases

- CIViC: https://civic.genome.wustl.edu
- IntOGen: http://www.intogen.org

#+BEGIN_CENTER
#+ATTR_LATEX: :width 0.3\textwidth
[[./images9/biology-of-cancer.jpg]]
#+END_CENTER
\vspace{0.01cm}
\scriptsize
http://www.amazon.com/The-Biology-Cancer-Robert-Weinberg/dp/0815340761
\normalsize

** Summary
\Large
- Demonstrate current validation work in bcbio
- Human build 38
- HLA typing
- Low frequency cancer calling
- Structural variations + prioritization
\vspace{1cm}
http://bcb.io
