#+title: Variant calling: tools, validation, genomes and outputs
#+author: Brad Chapman \\ Bioinformatics Core, Harvard Chan School \\ https://github.com/chapmanb/bcbio-nextgen \\ http://bcb.io \\ http://j.mp/bcbiolinks
#+date: 19 May 2015

#+OPTIONS: toc:nil H:2

#+startup: beamer
#+LaTeX_CLASS: beamer
#+latex_header: \usepackage{url}
#+latex_header: \usepackage{hyperref}
#+latex_header: \hypersetup{colorlinks=true}
#+BEAMER_THEME: default
#+BEAMER_COLOR_THEME: seahorse
#+BEAMER_INNER_THEME: rectangles

* Variant calling overview

** Outline

\Large
- \textbf{Variant calling made easy}
- Tools
- Validation
- Post-calling annotation
- Genomes and graphs
- Understanding outputs
- Automating everything -- bcbio

** Human whole genome sequencing

[[./images5/human_genome.png]]

\footnotesize
http://ensembl.org/Homo_sapiens/Location/Genome
\normalsize

** High throughput sequencing

[[./images5/reads.png]]

** Variant calling

[[./images5/SNV_calling.png]]

\footnotesize
http://en.wikipedia.org/wiki/SNV_calling_from_NGS_data
\normalsize

** Scale: exome to whole genome

[[./images5/exome_proportion.png]]

\footnotesize
https://www.flickr.com/photos/119980645@N06/
\normalsize

* Calling methods

** Outline

\Large
- Variant calling made easy
- \textbf{Tools}
- Validation
- Post-calling annotation
- Genomes and graphs
- Understanding outputs
- Automating everything -- bcbio

** SNPs and Indels

[[./images8/mutations.png]]

http://carolguze.com/text/442-2-mutations.shtml

** Structural variations

#+BEGIN_CENTER
#+ATTR_LATEX: :width .55\textwidth
[[./images8/svs.jpg]]
#+END_CENTER

\footnotesize
http://www.nature.com/nmeth/journal/v9/n2/full/nmeth.1858.html

** Germline population calling

#+ATTR_LATEX: :width .8\textwidth
[[./images8/Utah-Pedigree-1463-with-NA12878.png]]

\tiny
http://blog.goldenhelix.com/grudy/the-state-of-ngs-variant-calling-dont-panic/
\normalsize

** Genome Analysis Toolkit (GATK)

[[./images8/gatk.png]]

\vspace{0.5cm}

https://www.broadinstitute.org/gatk/

** GATK Best Practices

[[./images8/gatk-bp.png]]

\small
https://www.broadinstitute.org/gatk/guide/best-practices
\normalsize

** HaplotypeCaller

[[./images8/gatk-hc.png]]

\tiny
http://gatkforums.broadinstitute.org/discussion/5464/workshop-presentations-2015-uk-4-20-24
\normal

** Joint calling on large populations

[[./images8/gatk-joint.png]]

\tiny
http://gatkforums.broadinstitute.org/discussion/5464/workshop-presentations-2015-uk-4-20-24
\normal

** Support and availability

#+BEGIN_CENTER
[[./images8/gatk-license.png]]
#+END_CENTER

\vspace{0.5cm}

https://github.com/broadgsa

** FreeBayes

[[./images8/freebayes.png]]

https://github.com/ekg/freebayes

** Filtering -- hard cutoffs

[[./images8/hard-filters.png]]

\vspace{1cm}

http://bcb.io/2014/05/12/wgs-trio-variant-evaluation/

** Filtering -- Variant Quality Score Recalibration

#+ATTR_LATEX: :width .9\textwidth
[[./images8/gatk-vqsr.png]]

\tiny
http://gatkforums.broadinstitute.org/discussion/5464/workshop-presentations-2015-uk-4-20-24
\normal

** Cancer somatic calling

[[./images8/cancer-somatic.png]]

\vspace{0.5cm}
\scriptsize
http://www.nature.com/nmeth/journal/v10/n8/fig_tab/nmeth.2562_F1.html
\normalsize

** Cancer heterogeneity

[[./images8/Treatment_bottleneck.pdf]]

\small
http://en.wikipedia.org/wiki/Tumour_heterogeneity
\normalsize

** Not a solved problem

\Large
#+BEGIN_QUOTE
Four major genome centers predicted single-nucleotide variants (SNVs) for The
Cancer Genome Atlas (TCGA) lung cancer samples, but only 31.0% (1,667/5,380) of
SNVs were identified by all four.
#+END_QUOTE

\scriptsize
http://www.nature.com/nmeth/journal/vaop/ncurrent/full/nmeth.3407.html

** MuTect

\Large
- Broad GATK UnifiedGenotyper based
- SNP only

\vspace{1cm}
\small
https://www.broadinstitute.org/cancer/cga/mutect
\normalsize

** VarDict

\Large
- AstraZeneca
- SNP + Insertion/Deletions
- Works on very deep targeted data

\vspace{1cm}
\small
https://github.com/AstraZeneca-NGS/VarDictJava
\normalsize

* Validating calls

** Outline

\Large
- Variant calling made easy
- Tools
- \textbf{Validation}
- Post-calling annotation
- Genomes and graphs
- Understanding outputs
- Automating everything -- bcbio

** Reference materials

#+BEGIN_CENTER
#+ATTR_LATEX: :width .5\textwidth
[[./images/giab.png]]

#+ATTR_LATEX: :width .7\textwidth
[[./images7/ga4gh.png]]

#+ATTR_LATEX: :width .9\textwidth
[[./images7/dream_challenge.png]]

[[http://www.genomeinabottle.org/]]
http://ga4gh.org/#/benchmarking-team
https://www.synapse.org/#!Synapse:syn312572
#+END_CENTER

** Validate and compare caller performance

#+BEGIN_CENTER
#+ATTR_LATEX: :width 1.0\textwidth
[[./images7/cancer_validate.png]]
#+END_CENTER

http://bcb.io/2015/03/05/cancerval/


* Post-calling annotation and analysis

** Outline

\Large
- Variant calling made easy
- Tools
- Validation
- \textbf{Post-calling annotation}
- Genomes and graphs
- Understanding outputs
- Automating everything -- bcbio

** Effects prediction

#+ATTR_LATEX: :width 1.1\textwidth
[[./images8/vep-consequences.jpg]]

\scriptsize
http://www.ensembl.org/info/genome/variation/predicted_data.html

** Tools for effects predictions

\Large
- snpEff

\small
http://snpeff.sourceforge.net/

\vspace{0.5cm}

\Large
- Variant Effect Predictor (VEP) from Ensembl

\small
http://www.ensembl.org/info/docs/tools/vep/index.html

** Annotation and analysis -- GEMINI

[[./images7/gemini_overview.png]]

\small
https://github.com/arq5x/gemini
\normalsize

* Genome representations

** Outline

\Large
- Variant calling made easy
- Tools
- Validation
- Post-calling annotation
- \textbf{Genomes and graphs}
- Understanding outputs
- Automating everything -- bcbio

** Currently: GRCh37/hg19

#+ATTR_LATEX: :width .9\textwidth
[[./images8/GRCh37.jpg]]

\small
http://www.ncbi.nlm.nih.gov/books/NBK153600/?report=reader

** GRCh38 -- graph based, many more alternative loci

#+ATTR_LATEX: :width .9\textwidth
[[./images8/GRCh38.jpg]]

\footnotesize
http://www.slideshare.net/GenomeRef/transitioning-to-grch38

** GRCh38 -- advantage for variant calling

#+ATTR_LATEX: :width .9\textwidth
[[./images8/GRCh38-advantages.jpg]]

\footnotesize
http://www.slideshare.net/GenomeRef/transitioning-to-grch38

** Major histocompatibility complex (MHC) -- HLAs

[[./images8/MHC.png]]

\small
\vspace{1cm}
http://www.ebi.ac.uk/ipd/imgt/hla/ \\
\scriptsize
http://sciscogenetics.com/technology/human-leukocyte-antigen-complex/

** Alignment: bwa alternative allele support

#+ATTR_LATEX: :width .9\textwidth
[[./images8/bwa-alts.png]]

\small
https://github.com/lh3/bwa/blob/master/README-alt.md

** Genome graphs and variation

#+ATTR_LATEX: :width .9\textwidth
[[./images8/discovar-graphs.jpg]]

\footnotesize
http://www.nature.com/ng/journal/v46/n12/fig_tab/ng.3121_SF6.html
\normalsize

** vg -- tools for working with variant graphs

#+BEGIN_CENTER
#+ATTR_LATEX: :width .5\textwidth
[[./images8/vg-graphs.png]]
#+END_CENTER

\tiny
https://docs.google.com/presentation/d/1bbl2zY4qWQ0yYBHhoVuXb79HdgajRotIUa_VEn3kTpI
\normalsize

* Understanding outputs

** Outline

\Large
- Variant calling made easy
- Tools
- Validation
- Post-calling annotation
- Genomes and graphs
- \textbf{Understanding outputs}
- Automating everything -- bcbio

** VCF -- overview

#+ATTR_LATEX: :width 1.05\textwidth
[[./images8/vcf-overview.png]]

http://vcftools.sourceforge.net/VCF-poster.pdf

** VCF -- representations

[[./images8/vcf-representation.png]]

http://vcftools.sourceforge.net/VCF-poster.pdf

** Learning to read VCFs

\Large
- Step by step guide from Broad
\small
https://www.broadinstitute.org/gatk/guide/article?id=1268

\vspace{0.5cm}
\Large
- Specification
\small
http://samtools.github.io/hts-specs/

* bcbio
** Outline

\Large
- Variant calling made easy
- Tools
- Validation
- Post-calling annotation
- Genomes and graphs
- Understanding outputs
- \textbf{Automating everything -- bcbio}

** White box software

[[./images5/clear_box.jpg]]

** Overview

#+ATTR_LATEX: :width 1.0\textwidth
[[./images3/bcbio_nextgen_highlevel.png]]

\vspace{1cm}
https://github.com/chapmanb/bcbio-nextgen

** Uses

\Large
- Aligners: bwa-mem, novoalign, bowtie2
- Variantion: FreeBayes, GATK, VarDict, MuTecT, Scalpel, SnpEff, VEP, GEMINI,
  Lumpy, Delly, CNVkit
- RNA-seq: Tophat, STAR, cufflinks, HTSeq
- Quality control: fastqc, bamtools, RNA-SeQC
- Manipulation: bedtools, bcftools, biobambam, sambamba, samblaster, samtools,
  vcflib, vt
\normalsize

** Provides

\Large
- Community -- collected set of expertise
- Validation -- outputs + automated evaluation
- Scaling
- Ready to run parallel processing on AWS
- Local installation of tools and data
\normalsize

** Complex, rapidly changing baseline functionality

[[./images2/gatk_changes.png]]

** Quality differences between methods

#+ATTR_LATEX: :width .7\textwidth
[[./images/gcat_comparison.png]]

[[http://www.bioplanet.com/gcat]]

** Benefits of improved filtering

#+BEGIN_CENTER
#+ATTR_LATEX: :width 0.8\textwidth
[[./images7/freebayes_improve-syn3-tumor.png]]
#+END_CENTER

http://j.mp/cancervalpre

** Solution

#+BEGIN_CENTER
#+ATTR_LATEX: :width .5\textwidth
[[./images/community.png]]
#+END_CENTER

\scriptsize
[[http://www.amazon.com/Community-Structure-Belonging-Peter-Block/dp/1605092770]]
\normalsize

** Community: contribution

[[./images5/bcbio_github.png]]

[[https://github.com/chapmanb/bcbio-nextgen]]

* Conclusions
** Outline

\Large
- Variant calling made easy
- Tools
- Validation
- Post-calling annotation
- Genomes and graphs
- Understanding outputs
- Automating everything -- bcbio
